k8s/eks/yaml/tg-daemon-deployment.yml

```diff
@@ -0,0 +1,82 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: testground-daemon
+  labels:
+    app: testground-daemon
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: testground-daemon
+  template:
+    metadata:
+      labels:
+        app: testground-daemon
+    spec:
+      serviceAccountName: testground-daemon
+      hostPID: true
+      hostNetwork: true
+      dnsPolicy: ClusterFirstWithHostNet
+      nodeSelector:
+        testground.node.role.infra: "true"
+      containers:
+      - name: goproxy
+        image: public.ecr.aws/n6b0k8i7/testground-goproxy:2.0.2
+        ports:
+        - containerPort: 8081
+          hostPort: 8081
+        volumeMounts:
+          - name: envtoml
+            mountPath: /root/testground/.env.toml
+            subPath: .env.toml
+          - name: efs-pvc
+            mountPath: "/go"
+        resources:
+          requests:
+            memory: 700Mi
+            cpu: 500m
+          limits:
+            memory: 1000Mi
+            cpu: 700m
+      - name: testground-daemon
+        image: public.ecr.aws/n6b0k8i7/testground-daemon:eks_v1.0
+        imagePullPolicy: Always
+        env:
+        - name: REDIS_HOST
+          value: "testground-infra-redis-master.default.svc.cluster.local"
+        securityContext:
+          privileged: true
+        ports:
+        - containerPort: 8042
+          hostPort: 8042
+        volumeMounts:
+          - name: daemon-datadir
+            mountPath: "/root/testground/"
+          - name: efs-pvc
+            mountPath: "/efs"
+          - name: dockersock
+            mountPath: "/var/run/docker.sock"
+          - name: envtoml
+            mountPath: /root/testground/.env.toml
+            subPath: .env.toml
+        resources:
+          requests:
+            memory: 2048Mi
+            cpu: 2000m
+          limits:
+            memory: 4096Mi
```

Comment:
Explained the increase [here](https://github.com/testground/testground/issues/1454).

Reply:
Let's have this information inline. Also, it'd be useful to know what kind of scale these settings should enable.

Reply:
Added a comment in the yaml.
As explained in the issue, this scaling would enable the daemon/sidecar to process and support a test of more than 40 pods rather than crashing the test due to lack of hardware resources.

Reply:
The issue states that it was crashing with 40 pods. The solution doesn't mention anything about what the new limits are.

Reply:
Updated the solution with new limits.

Reply:
I meant, what is the maximum number of pods if we use these settings?

Reply:
Storm test with 300 instances seems to be passing. I have not looked at any metrics though so have no idea what limits that pushes.

Reply:
Yeah, 300 for 2 worker nodes should be within hardware limits.
The `storm` pods each require `cpu: 100m` and `memory:  100Mi`, so 1/10th of a cpu core/1 GB RAM.
Our default instances are `c5.4xlarge` with 16 cpu cores/32GB RAM. 
Having all this in mind, the rough estimate would be
~ 160 storm pods per worker node when it comes to cpu limits, or ~ 320 pods per worker node when it comes to RAM limits.

These limits you have linked for the daemon and sidecar have been successfully tested for up to 3000 (I think even 3200) pods, using 20 worker nodes in the `plan` nodegroup.

Reply:
How did you come up with the pod requirements?

Reply:
When you schedule a `storm` test for example and the pods get created, you will be able to issue:

`kubectl describe pod storm_pod_id`

and see what both the `limits` and `requests` are for the pods.

More info on pod resources can be found [here](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/).

Reply:
OK, this makes sense when it comes to the plan worker nodes. Thank you for the explanation 🙇 

I'm still wondering about the daemon though. It uses some baseline amount of resources on its own. But then whenever we add a new pod, it adds a strain on the daemon as well. It'd be interesting to know what's the resource consumption increase on the daemon when there's a new plan pod. I think it'll be next to impossible to estimate without metrics though.

Reply:
I'm going to resolve it given we don't have enough visibility atm. And we're going to follow up on it at some point in the future.